hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServerSource.java

```diff
@@ -571,6 +571,9 @@
   String RPC_SCAN_REQUEST_COUNT = "rpcScanRequestCount";
   String RPC_SCAN_REQUEST_COUNT_DESC =
       "Number of rpc scan requests this RegionServer has answered.";
+  String RPC_FULL_SCAN_REQUEST_COUNT = "rpcFullScanRequestCount";
+  String RPC_FULL_SCAN_REQUEST_COUNT_DESC =
+      "Number of rpc scan requests that were possible full region scans.";
```

Comment:
Full Region or full Table?

Reply:
`Looks good. Some nits in below. Main question is if this count has value? If so, how? Thanks.`
@saintstack  - the main reason was that we generally don't get an insight if the user has issued a scan without really setting a start and stop row. Mostly they want a specific set of rows in the scan but they generally miss setting the start and stop row and depend on filters. Though filters are an option but still deciding the rowkeys upfront to ensure we make use of the start and stop row is a good practice. If we have a metric to capture we can atleast indicate the pattern but whether it is really intentional or not may be something which as experts we can decide and inform the user. Also though some time it is intentional it might not be the actual business logic that is doing the scan but a sanity check that the user is running to do some verifications. that is not latency sensitive but still it will evict the cached data which is actually needed for the real latency sensitive queries. In such cases atleast we can suggest that the scan that is doing a full range/table scan can be made not to use the cache by setting `scan#setCaching(false)`